### ISSUE TITLE
TestListenTCPWithTOS failing on go tip

### ISSUE THREAD

**Issue by tsuna** [aristanetworks/goarista#15]
The first recorded failure was seen in [Job #179.2](https://travis-ci.org/aristanetworks/goarista/jobs/203918684), which built with `go version devel +689fa9c Tue Feb 21 18:43:02 2017 +0000 linux/amd64`.  Might be worthwhile taking a look at what's going on over there.

```
=== RUN   TestListenTCPWithTOS
--- FAIL: TestListenTCPWithTOS (0.00s)
panic: reflect: call of reflect.Value.Int on zero Value [recovered]
	panic: reflect: call of reflect.Value.Int on zero Value
goroutine 6 [running]:
testing.tRunner.func1(0xc42006a8f0)
	/home/travis/.gimme/versions/go/src/testing/testing.go:624 +0x264
panic(0x55f2a0, 0xc42000ce40)
	/home/travis/.gimme/versions/go/src/runtime/panic.go:489 +0x25a
reflect.Value.Int(0x0, 0x0, 0x0, 0x58fb4e)
	/home/travis/.gimme/versions/go/src/reflect/value.go:902 +0xb5
github.com/aristanetworks/goarista/dscp.listenTCPWithTOS(0xc420010f00, 0x28, 0x570d00, 0x1, 0xc420010f00)
	/home/travis/gopath/src/github.com/aristanetworks/goarista/dscp/listen_unix.go:22 +0x14a
github.com/aristanetworks/goarista/dscp.ListenTCPWithTOS(0xc420010f00, 0xc420010f28, 0xc4200151d0, 0x10, 0x10)
	/home/travis/gopath/src/github.com/aristanetworks/goarista/dscp/listen.go:16 +0x34
github.com/aristanetworks/goarista/dscp_test.testListenTCPWithTOS(0xc42006a8f0, 0x590478, 0x9)
	/home/travis/gopath/src/github.com/aristanetworks/goarista/dscp/listen_test.go:24 +0xcf
github.com/aristanetworks/goarista/dscp_test.TestListenTCPWithTOS(0xc42006a8f0)
	/home/travis/gopath/src/github.com/aristanetworks/goarista/dscp/listen_test.go:15 +0x40
testing.tRunner(0xc42006a8f0, 0x599468)
	/home/travis/.gimme/versions/go/src/testing/testing.go:659 +0x98
created by testing.(*T).Run
	/home/travis/.gimme/versions/go/src/testing/testing.go:701 +0x2da
FAIL	github.com/aristanetworks/goarista/dscp	0.018s
```

**Comment by tsuna** [comment]
Ah, that's because the `sysFD` struct that we access somewhat illegally has changed in https://go-review.googlesource.com/#/c/36799/7/src/net/fd_unix.go@19

So we'll need to tweak our kludge a bit to take into account the new struct definition.

**Comment by aaronbee** [comment]
Fixed with c41ed3986faaeb0aeed2bd98e92468f8ea317954
